app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [{"line": error["line_number"],
+            "column": error["column_number"], "message": error["text"], "name":
+                       error["code"], "source": "flake8"} for error in errors],
+            "path": file_path, "status": "failed" if errors else "passed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{"errors": [{"line": error["line_number"],
+                         "column": error["column_number"],
+             "message": error["text"], "name": error["code"],
+            "source": "flake8"} for error in errors], "path": file_path,
+             "status": "failed" if errors else "passed"} for file_path,
+            errors in linter_report.items()]
```

Comment:
same as above, try to use `format_single_linter_file`instead of duplicating code here.